modules/client-handler/src/main/java/org/apache/ignite/client/handler/JdbcQueryEventHandlerImpl.java

```diff
@@ -514,6 +510,10 @@ <T> CompletableFuture<T> doInSession(SessionAwareAction<T> action) {
 
         private SessionId recreateSession(@Nullable SessionId expectedSessionId) {
             synchronized (mux) {
+                if (closed) {
+                    throw new IgniteInternalException(CONNECTION_ERR, "Connection is closed");
```

Comment:
it's better to cover this with tests to see what will happen

Reply:
I agree, but creating a good test for such a scenario seems not an easy task.
Added something like a test called `sessionCreationAfterCloseRegistry` to cover this case.